### ISSUE TITLE
does not play properly on ios

### ISSUE THREAD

**Issue by devappsstudio** [vokhuyetOz/react-native-concurrent-sound#15]
I have a hook for loading tracks

```
const usePlayer = (volume: number | undefined, effects: number | undefined) => {
  useEffect(() => {
    const eventListener = ConcurrentSoundEvent.addListener(
      'OnSoundEnd',
      (event: any) => {
        console.log('OnSoundEnd', event);
      },
    );

    // Removes the listener once unmounted
    return () => {
      eventListener.remove();
    };
  }, []);

  const startPlayBg = async () => {
    const duration = await load({
      key: 'bg_music',
      uri: require('../assets/audio/bg_music.mp3'),
      loop: true,
      volume: volume,
    });
    console.log('Load and Play bg_music assets', duration);
    play({
      key: 'bg_music',
    });
  };

  const loadBallSound = async () => {
    await load({
      key: 'ball_sound',
      uri: require('../assets/audio/ball_sound.mp3'),
      volume: effects,
    });
  };

  const loadExplosionSound = async () => {
    await load({
      key: 'explosion_sound',
      uri: require('../assets/audio/explosion_sound.mp3'),
      volume: effects,
    });
  };

  useEffect(() => {
    startPlayBg();
    loadBallSound();
    loadExplosionSound();
  }, []);

  useEffect(() => {
    setVolume({
      key: 'bg_music',
      to: volume ?? 0,
    });
  }, [volume]);

  useEffect(() => {
    setVolume({
      key: 'ball_sound',
      to: effects ?? 0,
    });

    setVolume({
      key: 'explosion_sound',
      to: effects ?? 0,
    });
  }, [effects]);
};
```
Than I use 

```
const callBallSound = () => {
  play({
    key: 'ball_sound',
  });
};
```
 
to run sound in specific places.

For android (device and emulator) it works great, sound runs every time it calls. But for ios (device and emulator) sound runs only for first time it calls.

Here is an example of video. "ball_sound" should run for each collision of ball. For android it works. But for ios "ball_sound" runs only for first collision.

https://github.com/user-attachments/assets/722d2c45-adf2-4a13-95c8-a327c67c145f

Is there any way to fix this?

**Comment by vokhuyetOz** [comment]
hmm, it is not happen on my side
let me try to reproduce it

**Comment by vokhuyetOz** [comment]
and which version of library are you using?

**Comment by devappsstudio** [comment]
here are my dependencies
```
 "dependencies": {
    "@react-native-async-storage/async-storage": "^2.1.0",
    "@react-navigation/elements": "^2.0.0",
    "@react-navigation/native": "^6.1.18",
    "@react-navigation/native-stack": "^6.11.0",
    "@react-navigation/stack": "^6.4.1",
    "@shopify/react-native-skia": "^1.4.2",
    "@vokhuyet/react-native-concurrent-sound": "^0.1.0",
    "react": "18.3.1",
    "react-native": "0.75.4",
    "react-native-gesture-handler": "^2.20.0",
    "react-native-reanimated": "^3.15.5",
    "react-native-safe-area-context": "^4.11.0",
    "react-native-screens": "^3.34.0",
    "react-native-splash-screen": "^3.3.0",
    "zustand": "^5.0.0"
  }
```
And here are two more things:
1. After each collision I see in console "OnSoundEnd {"key": "ball_sound"}" in progression after each next collision. After first collision I see in console new "OnSoundEnd {"key": "ball_sound"}" one time; after second collision - two more times in a row; after third collision - three more times in a row ...
2. Adding seek({key: 'ball_sound', to: 0}) to callBallSound 
```
const callBallSound = () => {
  play({
    key: 'ball_sound',
  });
  seek({key: 'ball_sound', to: 0});
}
```
somehow solves this issue. But in this case there are moments when the track does not play.

**Comment by vokhuyetOz** [comment]
sorry for my late
actually, it is not a bug
it is default behavior of iOS. 
on android player auto reset to 0 after finished, on iOS it is not

i updated some custom code to synchronized behavior of both Android and iOS

code is published on `0.1.1`
